### DISCUSSION TITLE
Reference for magnetic variation

### DISCUSSION THREAD

**Discussion by hudkinsnoah** [JSBSim-Team/jsbsim#1045] [Q&A]
I just updated my version (1.1.6 -> 1.2.0) of JSBSim and one of the references that I was using doesn't seem to be valid any more for getting the magnetic variation. The reference that I had been using before was ``fcs/mag-var``. I have looked through the catalog of references but don't see any that might be what I am looking for. Any help would be appreciated!

EDIT: I am looking for this to account for my heading being incorrect on my connected avionics when compared to the track. The avionics automatically calculate the magnetic variation but JSBSim(with any of the heading references I have tried using) isn't so values aren't lining up

**Comment by bcoconni** [comment]
Hi @hudkinsnoah,

The property `fcs/mag-var` is not a property that JSBSim provides. This property must be part of the FDM you are using and may have been removed at some point.

**Comment by seanmcleod** [comment]
Although JSBSim doesn't have a `mag-var` property we do have code implementing a geomagnetic model in the JSBSim, i.e. in - https://github.com/JSBSim-Team/jsbsim/blob/3d95aeb751e6ddd11b1007571b86662cd41612d8/src/simgear/magvar/coremag.cxx

And a magnetometer sensor that uses the geomagnetic model - https://github.com/JSBSim-Team/jsbsim/blob/3d95aeb751e6ddd11b1007571b86662cd41612d8/src/models/flight_control/FGMagnetometer.cpp#L121
